source/compiler-core/slang-lexer.cpp

```diff
@@ -829,17 +829,47 @@ FloatingPointLiteralValue getFloatingPointLiteralValue(
     return value;
 }
 
-static void _lexStringLiteralBody(Lexer* lexer, char quote)
+IntegerLiteralValue getCharLiteralValue(Token const& token)
 {
+    const UnownedStringSlice content = token.getContent();
```

Comment:
what if the token content contains escape sequences?

e.g. `'\t'`?

Reply:
May want to use `StringEscapeUtil::unquote()` here.

Reply:
That's a really good point. I've now added an escaped char in the test and unquoted appropriately. It had to be done in a bit of a roundabout way; `StringEscapeUtil::unquote()` doesn't unescape, and `StringEscapeUtil::unescapeString` assumes that the string starts with `"` and not `'`.